osh/builtin_printf.py

```diff
@@ -252,14 +314,31 @@ def Run(self, cmd_val):
                 s = '%x' % d
               elif typ == 'X':
                 s = '%X' % d
+            elif part.type.id == Id.Format_Time:
+              # set timezone
+              tzcell = self.mem.GetCell('TZ')
+              if tzcell and tzcell.exported and tzcell.val.tag_() == value_e.Str:
+                tzval = cast(value__Str, tzcell.val)
+                os.environ['TZ'] = tzval.s
+              elif 'TZ' in os.environ:
+                del os.environ['TZ']
+              time.tzset()
+
+              if d == -1: # now
+                d = None
+              elif d == -2: # shell start time
+                d = shell_start_time
+              s = time.strftime(typ[1:-2], time.localtime(d));
```

Comment:
no semicolon needed

Also where does `-2` come from?  I'm having trouble following that.  May need more comments.

Reply:
I removed the semicolon and added comments in dcf99bd. Also, I changed to explicitly set `d = time.time()` (not relying on the default behavior of `time.localtime(None)`).